teamcity: failed tests on master: acceptance/TestDockerCLI, acceptance/TestDockerCLI/test_missing_log_output.tcl

The following tests appear to have failed:

[#332941](https://teamcity.cockroachdb.com/viewLog.html?buildId=332941):

```
--- FAIL: acceptance/TestDockerCLI (167.970s)
test_log_scope.go:78: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestDockerCLI870265889
test_log_scope.go:61: use -show-logs to present logs inline
------- Stdout: -------
xenial-20170214: Pulling from library/ubuntu
Digest: sha256:dd7808d8792c9841d0b460122f1acf0a2dd1f56404f8d1e56298048885e45535
Status: Image is up to date for ubuntu:xenial-20170214
20170423-1100: Pulling from cockroachdb/postgres-test
Digest: sha256:1fd869a88a4316aa1e464501fdff364aef43512ca8aa12914ebc0790f37a58c6
Status: Image is up to date for cockroachdb/postgres-test:20170423-1100
  File: ���/cockroach/cockroach���
  Size: 43072808  	Blocks: 84136      IO Block: 4096   regular file
Device: 801h/2049d	Inode: 157029      Links: 1
Access: (0755/-rwxr-xr-x)  Uid: ( 1001/ UNKNOWN)   Gid: ( 1001/ UNKNOWN)
Access: 2017-08-23 22:01:34.928504299 +0000
Modify: 2017-08-23 22:01:05.555315331 +0000
Change: 2017-08-23 22:01:05.591316789 +0000
 Birth: -
--- FAIL: acceptance/TestDockerCLI/test_missing_log_output.tcl (12.080s)
cli_test.go:87: non-zero exit code: 1
------- Stdout: -------
xenial-20170214: Pulling from library/ubuntu
Digest: sha256:dd7808d8792c9841d0b460122f1acf0a2dd1f56404f8d1e56298048885e45535
Status: Image is up to date for ubuntu:xenial-20170214
20170423-1100: Pulling from cockroachdb/postgres-test
Digest: sha256:1fd869a88a4316aa1e464501fdff364aef43512ca8aa12914ebc0790f37a58c6
Status: Image is up to date for cockroachdb/postgres-test:20170423-1100
expect version 5.45
argv[0] = expect  argv[1] = -d  argv[2] = -f  argv[3] = /go/src/github.com/cockroachdb/cockroach/cli/interactive_tests/test_missing_log_output.tcl  argv[4] = /cockroach/cockroach  
set argc 1
set argv0 "/go/src/github.com/cockroachdb/cockroach/cli/interactive_tests/test_missing_log_output.tcl"
set argv "/cockroach/cockroach"
executing commands from command file /go/src/github.com/cockroachdb/cockroach/cli/interactive_tests/test_missing_log_output.tcl
system(rm -f cockroach_sql_history) = 2
trap: setting up signal 2 ("SIGINT")
trap: setting up signal 15 ("SIGTERM")
parent: waiting for sync byte
spawn /bin/bash
parent: telling child to go ahead
parent: now unsynchronized from child
spawn: returns {10}
send: sending "PS1=':''/# '\r" to { exp5 }

expect: does "" (spawn_id exp5) match glob pattern ":/# "? no

expect: does "PS1=':''/# '\r\n" (spawn_id exp5) match glob pattern ":/# "? no
PS1=':''/# '

expect: does "PS1=':''/# '\r\nroot@6fad77e0416e:/# PS1=':''/" (spawn_id exp5) match glob pattern ":/# "? yes
expect: set expect_out(0,string) ":/# "
expect: set expect_out(spawn_id) "exp5"
expect: set expect_out(buffer) "PS1=':''/# '\r\nroot@6fad77e0416e:/# "
root@6fad77e0416e:/# PS1=':''/
.170823 22:03:26.090495909 EXPECT TEST: START TEST: Check that a server encountering a fatal error when not logging to stderr shows the fatal error.
system(echo; echo $(date '+.%y%m%d %H:%M:%S.%N') EXPECT TEST: 'START TEST: Check that a server encountering a fatal error when not logging to stderr shows the fatal error.' | tee -a logs/expect-cmd.log) = 2
system(find logs -exec chmod a+rw '{}' \;) = 2
send: sending "/cockroach/cockroach start -s=path=logs/db --insecure\r" to { exp5 }

expect: does "PS1=':''/" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "PS1=':''/# '\r\n:/# /cockroach" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no
# '

expect: does "PS1=':''/# '\r\n:/# /cockroach/cockr" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "PS1=':''/# '\r\n:/# /cockroach/cockroach" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "PS1=':''/# '\r\n:/# /cockroach/cockroach sta" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "PS1=':''/# '\r\n:/# /cockroach/cockroach start " (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "PS1=':''/# '\r\n:/# /cockroach/cockroach start -s=" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "PS1=':''/# '\r\n:/# /cockroach/cockroach start -s=pa" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "PS1=':''/# '\r\n:/# /cockroach/cockroach start -s=path=" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "PS1=':''/# '\r\n:/# /cockroach/cockroach start -s=path=lo" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "PS1=':''/# '\r\n:/# /cockroach/cockroach start -s=path=logs" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "PS1=':''/# '\r\n:/# /cockroach/cockroach start -s=path=logs/" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "PS1=':''/# '\r\n:/# /cockroach/cockroach start -s=path=logs/db " (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "PS1=':''/# '\r\n:/# /cockroach/cockroach start -s=path=logs/db --" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "PS1=':''/# '\r\n:/# /cockroach/cockroach start -s=path=logs/db --in" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "PS1=':''/# '\r\n:/# /cockroach/cockroach start -s=path=logs/db --inse" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "PS1=':''/# '\r\n:/# /cockroach/cockroach start -s=path=logs/db --insecu" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "PS1=':''/# '\r\n:/# /cockroach/cockroach start -s=path=logs/db --insecur" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "PS1=':''/# '\r\n:/# /cockroach/cockroach start -s=path=logs/db --insecure" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no
:/# /cockroach/cockroach start -s=path=logs/db --insecure

expect: does "PS1=':''/# '\r\n:/# /cockroach/cockroach start -s=path=logs/db --insecure\r\n" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no
*
* WARNING: RUNNING IN INSECURE MODE!
* 
* - Your cluster is open for any client that can access <all your IP addresses>.
* - Any user, even root, can log in without providing a password.

expect: does "PS1=':''/# '\r\n:/# /cockroach/cockroach start -s=path=logs/db --insecure\r\n*\r\n* WARNING: RUNNING IN INSECURE MODE!\r\n* \r\n* - Your cluster is open for any client that can access <all your IP addresses>.\r\n* - Any user, even root, can log in without providing a password.\r\n* - Any user, connecting as root, can read or write any data in your cluster.\r\n* - There is no network encryption nor authentication, and thus no confidentiality.\r\n* \r\n* Check out how to secure your cluster: https://www.cockroachlabs.com/docs/stable/secure-a-cluster.html\r\n*\r\n" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no
* - Any user, connecting as root, can read or write any data in your cluster.
* - There is no network encryption nor authentication, and thus no confidentiality.
* 
* Check out how to secure your cluster: https://www.cockroachlabs.com/docs/stable/secure-a-cluster.html
*
CockroachDB node starting at 2017-08-23 22:03:27.996348447 +0000 UTC (took 1.9s)
build:      CCL 42e183b @ 2017/08/23 22:00:38 (go1.8.3)
admin:      http://6fad77e0416e:8080
sql:        postgresql://root@6fad77e0416e:26257?application_name=cockroach&sslmode=disable
logs:       /logs/db/logs
store[0]:   path=/logs/db
status:     initialized new cluster
clusterID:  ce1f6e25-9525-4e77-879b-34329c0cc6d2
nodeID:     1

expect: does "PS1=':''/# '\r\n:/# /cockroach/cockroach start -s=path=logs/db --insecure\r\n*\r\n* WARNING: RUNNING IN INSECURE MODE!\r\n* \r\n* - Your cluster is open for any client that can access <all your IP addresses>.\r\n* - Any user, even root, can log in without providing a password.\r\n* - Any user, connecting as root, can read or write any data in your cluster.\r\n* - There is no network encryption nor authentication, and thus no confidentiality.\r\n* \r\n* Check out how to secure your cluster: https://www.cockroachlabs.com/docs/stable/secure-a-cluster.html\r\n*\r\nCockroachDB node starting at 2017-08-23 22:03:27.996348447 +0000 UTC (took 1.9s)\r\nbuild:      CCL 42e183b @ 2017/08/23 22:00:38 (go1.8.3)\r\nadmin:      http://6fad77e0416e:8080\r\nsql:        postgresql://root@6fad77e0416e:26257?application_name=cockroach&sslmode=disable\r\nlogs:       /logs/db/logs\r\nstore[0]:   path=/logs/db\r\nstatus:     initialized new cluster\r\nclusterID:  ce1f6e25-9525-4e77-879b-34329c0cc6d2\r\nnodeID:     1\r\n" (spawn_id exp5) match glob pattern "CockroachDB node starting"? yes
expect: set expect_out(0,string) "CockroachDB node starting"
expect: set expect_out(spawn_id) "exp5"
expect: set expect_out(buffer) "PS1=':''/# '\r\n:/# /cockroach/cockroach start -s=path=logs/db --insecure\r\n*\r\n* WARNING: RUNNING IN INSECURE MODE!\r\n* \r\n* - Your cluster is open for any client that can access <all your IP addresses>.\r\n* - Any user, even root, can log in without providing a password.\r\n* - Any user, connecting as root, can read or write any data in your cluster.\r\n* - There is no network encryption nor authentication, and thus no confidentiality.\r\n* \r\n* Check out how to secure your cluster: https://www.cockroachlabs.com/docs/stable/secure-a-cluster.html\r\n*\r\nCockroachDB node starting"
Error: driver: bad connection
Failed running "sql"
system(/cockroach/cockroach sql --insecure -e "select crdb_internal.force_log_fatal('helloworld')" || true) = 2

expect: does " at 2017-08-23 22:03:27.996348447 +0000 UTC (took 1.9s)\r\nbuild:      CCL 42e183b @ 2017/08/23 22:00:38 (go1.8.3)\r\nadmin:      http://6fad77e0416e:8080\r\nsql:        postgresql://root@6fad77e0416e:26257?application_name=cockroach&sslmode=disable\r\nlogs:       /logs/db/logs\r\nstore[0]:   path=/logs/db\r\nstatus:     initialized new cluster\r\nclusterID:  ce1f6e25-9525-4e77-879b-34329c0cc6d2\r\nnodeID:     1\r\n" (spawn_id exp5) match glob pattern "\r\nF"? no

*
* ERROR: [client=[::1]:45975,user=root,n1] Reported as error 7291d81af3544e3bb2d520d82ec60fb3
*
F170823 22:03:28.376782 618 sql/parser/builtins.go:1840  [client=[::1]:45975,user=root,n1] helloworld
goroutine 618 [running]:
github.com/cockroachdb/cockroach/pkg/util/log.getStacks(0x200, 0x16753f99, 0x2f195e0, 0x10)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:824 +0xa7
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).outputLogEntry(0x2f1b2c0, 0xc400000004, 0x277a2d6, 0x16, 0x730, 0xc420e59e90, 0x2c)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:729 +0x576
github.com/cockroachdb/cockroach/pkg/util/log.addStructured(0x7f5625754910, 0xc420c42480, 0x7f5600000004, 0x2, 0x0, 0x0, 0xc42129a180, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/structured.go:161 +0x487
github.com/cockroachdb/cockroach/pkg/util/log.logDepth(0x7f5625754910, 0xc420c42480, 0x1, 0xc400000004, 0x0, 0x0, 0xc42129a180, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:52 +0x8c
github.com/cockroachdb/cockroach/pkg/util/log.Fatal(0x7f5625754910, 0xc420c42480, 0xc42129a180, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:155 +0x76
github.com/cockroachdb/cockroach/pkg/sql/parser.glob..func142(0xc42127c248, 0xc4203304c0, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/parser/builtins.go:1840 +0x117
github.com/cockroachdb/cockroach/pkg/sql/parser.(*FuncExpr).Eval(0xc420d20d80, 0xc42127c248, 0x1, 0xc4203304b0, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/parser/eval.go:2750 +0x232
github.com/cockroachdb/cockroach/pkg/sql.(*renderNode).renderRow(0xc4212327e0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/render.go:559 +0x77
github.com/cockroachdb/cockroach/pkg/sql.(*renderNode).Next(0xc4212327e0, 0x7f5625754910, 0xc420c42480, 0xc42127c1f8, 0xc4212327e0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/render.go:112 +0xc7
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execClassic(0xc420678900, 0xc42127c1f8, 0x2a75680, 0xc4212327e0, 0x2a7f040, 0xc420da8300, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1713 +0x272
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmt(0xc420678900, 0x2a76c40, 0xc420e59bf0, 0x0, 0x0, 0x0, 0x14dd98c1a27ac222, 0x1, 0xc420f69f80, 0xc42127c1f8, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1843 +0x62b
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmtInOpenTxn(0xc420678900, 0xc42127c000, 0x2a76c40, 0xc420e59bf0, 0x0, 0x0, 0x0, 0x14dd98c1a27ac222, 0x1, 0xc420f69f80, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1556 +0x6ce
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmtsInCurrentTxn(0xc420678900, 0xc42127c000, 0xc420c42400, 0x1, 0x1, 0x0, 0x1, 0x0, 0x2a76fc0, 0xc420da8300, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1127 +0x8c1
github.com/cockroachdb/cockroach/pkg/sql.runTxnAttempt(0xc420678900, 0xc42127c000, 0xc420c42400, 0x1, 0x1, 0x0, 0x1, 0xc420330288, 0x1, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1013 +0xc9
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execParsed.func1(0x7f5625754910, 0xc420c42480, 0xc420e5c160, 0xc420330288, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:821 +0x294
github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Exec(0xc420e5c160, 0x7f5625754910, 0xc420c42480, 0x101, 0xc420da4500, 0x201a852, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:722 +0xa8
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execParsed(0xc420678900, 0xc42127c000, 0xc420c42400, 0x1, 0x1, 0x0, 0x0, 0xc42001d900, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:842 +0x47f
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).execRequest(0xc420678900, 0xc42127c000, 0xc42127e000, 0x32, 0x0, 0x0, 0x32, 0x1000)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:727 +0x482
expect: does " at 2017-08-23 22:03:27.996348447 +0000 UTC (took 1.9s)\r\nbuild:      CCL 42e183b @ 2017/08/23 22:00:38 (go1.8.3)\r\nadmin:      http://6fad77e0416e:8080\r\nsql:        postgresql://root@6fad77e0416e:26257?application_name=cockroach&sslmode=disable\r\nlogs:       /logs/db/logs\r\nstore[0]:   path=/logs/db\r\nstatus:     initialized new cluster\r\nclusterID:  ce1f6e25-9525-4e77-879b-34329c0cc6d2\r\nnodeID:     1\r\n*\r\n* ERROR: [client=[::1]:45975,user=root,n1] Reported as error 7291d81af3544e3bb2d520d82ec60fb3\r\n*\r\nF170823 22:03:28.376782 618 sql/parser/builtins.go:1840  [client=[::1]:45975,user=root,n1] helloworld\r\ngoroutine 618 [running]:\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.getStacks(0x200, 0x16753f99, 0x2f195e0, 0x10)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:824 +0xa7\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).outputLogEntry(0x2f1b2c0, 0xc400000004, 0x277a2d6, 0x16, 0x730, 0xc420e59e90, 0x2c)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:729 +0x576\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.addStructured(0x7f5625754910, 0xc420c42480, 0x7f5600000004, 0x2, 0x0, 0x0, 0xc42129a180, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/structured.go:161 +0x487\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.logDepth(0x7f5625754910, 0xc420c42480, 0x1, 0xc400000004, 0x0, 0x0, 0xc42129a180, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:52 +0x8c\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.Fatal(0x7f5625754910, 0xc420c42480, 0xc42129a180, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:155 +0x76\r\ngithub.com/cockroachdb/cockroach/pkg/sql/parser.glob..func142(0xc42127c248, 0xc4203304c0, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/parser/builtins.go:1840 +0x117\r\ngithub.com/cockroachdb/cockroach/pkg/sql/parser.(*FuncExpr).Eval(0xc420d20d80, 0xc42127c248, 0x1, 0xc4203304b0, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/parser/eval.go:2750 +0x232\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*renderNode).renderRow(0xc4212327e0, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/render.go:559 +0x77\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*renderNode).Next(0xc4212327e0, 0x7f5625754910, 0xc420c42480, 0xc42127c1f8, 0xc4212327e0, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/render.go:112 +0xc7\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execClassic(0xc420678900, 0xc42127c1f8, 0x2a75680, 0xc4212327e0, 0x2a7f040, 0xc420da8300, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1713 +0x272\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmt(0xc420678900, 0x2a76c40, 0xc420e59bf0, 0x0, 0x0, 0x0, 0x14dd98c1a27ac222, 0x1, 0xc420f69f80, 0xc42127c1f8, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1843 +0x62b\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmtInOpenTxn(0xc420678900, 0xc42127c000, 0x2a76c40, 0xc420e59bf0, 0x0, 0x0, 0x0, 0x14dd98c1a27ac222, 0x1, 0xc420f69f80, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1556 +0x6ce\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmtsInCurrentTxn(0xc420678900, 0xc42127c000, 0xc420c42400, 0x1, 0x1, 0x0, 0x1, 0x0, 0x2a76fc0, 0xc420da8300, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1127 +0x8c1\r\ngithub.com/cockroachdb/cockroach/pkg/sql.runTxnAttempt(0xc420678900, 0xc42127c000, 0xc420c42400, 0x1, 0x1, 0x0, 0x1, 0xc420330288, 0x1, 0x0, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1013 +0xc9\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execParsed.func1(0x7f5625754910, 0xc420c42480, 0xc420e5c160, 0xc420330288, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:821 +0x294\r\ngithub.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Exec(0xc420e5c160, 0x7f5625754910, 0xc420c42480, 0x101, 0xc420da4500, 0x201a852, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:722 +0xa8\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execParsed(0xc420678900, 0xc42127c000, 0xc420c42400, 0x1, 0x1, 0x0, 0x0, 0xc42001d900, 0x2)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:842 +0x47f\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execRequest(0xc420678900, 0xc42127c000, 0xc42127e000, 0x32, 0x0, 0x0, 0x32, 0x1000)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:727 +0x482\r\ngith" (spawn_id exp5) match glob pattern "\r\nF"? yes
expect: set expect_out(0,string) "\r\nF"
expect: set expect_out(spawn_id) "exp5"
expect: set expect_out(buffer) " at 2017-08-23 22:03:27.996348447 +0000 UTC (took 1.9s)\r\nbuild:      CCL 42e183b @ 2017/08/23 22:00:38 (go1.8.3)\r\nadmin:      http://6fad77e0416e:8080\r\nsql:        postgresql://root@6fad77e0416e:26257?application_name=cockroach&sslmode=disable\r\nlogs:       /logs/db/logs\r\nstore[0]:   path=/logs/db\r\nstatus:     initialized new cluster\r\nclusterID:  ce1f6e25-9525-4e77-879b-34329c0cc6d2\r\nnodeID:     1\r\n*\r\n* ERROR: [client=[::1]:45975,user=root,n1] Reported as error 7291d81af3544e3bb2d520d82ec60fb3\r\n*\r\nF"

expect: does "170823 22:03:28.376782 618 sql/parser/builtins.go:1840  [client=[::1]:45975,user=root,n1] helloworld\r\ngoroutine 618 [running]:\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.getStacks(0x200, 0x16753f99, 0x2f195e0, 0x10)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:824 +0xa7\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).outputLogEntry(0x2f1b2c0, 0xc400000004, 0x277a2d6, 0x16, 0x730, 0xc420e59e90, 0x2c)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:729 +0x576\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.addStructured(0x7f5625754910, 0xc420c42480, 0x7f5600000004, 0x2, 0x0, 0x0, 0xc42129a180, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/structured.go:161 +0x487\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.logDepth(0x7f5625754910, 0xc420c42480, 0x1, 0xc400000004, 0x0, 0x0, 0xc42129a180, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:52 +0x8c\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.Fatal(0x7f5625754910, 0xc420c42480, 0xc42129a180, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:155 +0x76\r\ngithub.com/cockroachdb/cockroach/pkg/sql/parser.glob..func142(0xc42127c248, 0xc4203304c0, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/parser/builtins.go:1840 +0x117\r\ngithub.com/cockroachdb/cockroach/pkg/sql/parser.(*FuncExpr).Eval(0xc420d20d80, 0xc42127c248, 0x1, 0xc4203304b0, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/parser/eval.go:2750 +0x232\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*renderNode).renderRow(0xc4212327e0, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/render.go:559 +0x77\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*renderNode).Next(0xc4212327e0, 0x7f5625754910, 0xc420c42480, 0xc42127c1f8, 0xc4212327e0, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/render.go:112 +0xc7\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execClassic(0xc420678900, 0xc42127c1f8, 0x2a75680, 0xc4212327e0, 0x2a7f040, 0xc420da8300, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1713 +0x272\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmt(0xc420678900, 0x2a76c40, 0xc420e59bf0, 0x0, 0x0, 0x0, 0x14dd98c1a27ac222, 0x1, 0xc420f69f80, 0xc42127c1f8, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1843 +0x62b\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmtInOpenTxn(0xc420678900, 0xc42127c000, 0x2a76c40, 0xc420e59bf0, 0x0, 0x0, 0x0, 0x14dd98c1a27ac222, 0x1, 0xc420f69f80, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1556 +0x6ce\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmtsInCurrentTxn(0xc420678900, 0xc42127c000, 0xc420c42400, 0x1, 0x1, 0x0, 0x1, 0x0, 0x2a76fc0, 0xc420da8300, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1127 +0x8c1\r\ngithub.com/cockroachdb/cockroach/pkg/sql.runTxnAttempt(0xc420678900, 0xc42127c000, 0xc420c42400, 0x1, 0x1, 0x0, 0x1, 0xc420330288, 0x1, 0x0, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1013 +0xc9\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execParsed.func1(0x7f5625754910, 0xc420c42480, 0xc420e5c160, 0xc420330288, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:821 +0x294\r\ngithub.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Exec(0xc420e5c160, 0x7f5625754910, 0xc420c42480, 0x101, 0xc420da4500, 0x201a852, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:722 +0xa8\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execParsed(0xc420678900, 0xc42127c000, 0xc420c42400, 0x1, 0x1, 0x0, 0x0, 0xc42001d900, 0x2)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:842 +0x47f\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execRequest(0xc420678900, 0xc42127c000, 0xc42127e000, 0x32, 0x0, 0x0, 0x32, 0x1000)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:727 +0x482\r\ngith" (spawn_id exp5) match glob pattern "helloworld"? yes
expect: set expect_out(0,string) "helloworld"
expect: set expect_out(spawn_id) "exp5"
expect: set expect_out(buffer) "170823 22:03:28.376782 618 sql/parser/builtins.go:1840  [client=[::1]:45975,user=root,n1] helloworld"

expect: does "\r\ngoroutine 618 [running]:\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.getStacks(0x200, 0x16753f99, 0x2f195e0, 0x10)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:824 +0xa7\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).outputLogEntry(0x2f1b2c0, 0xc400000004, 0x277a2d6, 0x16, 0x730, 0xc420e59e90, 0x2c)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:729 +0x576\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.addStructured(0x7f5625754910, 0xc420c42480, 0x7f5600000004, 0x2, 0x0, 0x0, 0xc42129a180, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/structured.go:161 +0x487\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.logDepth(0x7f5625754910, 0xc420c42480, 0x1, 0xc400000004, 0x0, 0x0, 0xc42129a180, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:52 +0x8c\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.Fatal(0x7f5625754910, 0xc420c42480, 0xc42129a180, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:155 +0x76\r\ngithub.com/cockroachdb/cockroach/pkg/sql/parser.glob..func142(0xc42127c248, 0xc4203304c0, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/parser/builtins.go:1840 +0x117\r\ngithub.com/cockroachdb/cockroach/pkg/sql/parser.(*FuncExpr).Eval(0xc420d20d80, 0xc42127c248, 0x1, 0xc4203304b0, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/parser/eval.go:2750 +0x232\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*renderNode).renderRow(0xc4212327e0, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/render.go:559 +0x77\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*renderNode).Next(0xc4212327e0, 0x7f5625754910, 0xc420c42480, 0xc42127c1f8, 0xc4212327e0, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/render.go:112 +0xc7\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execClassic(0xc420678900, 0xc42127c1f8, 0x2a75680, 0xc4212327e0, 0x2a7f040, 0xc420da8300, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1713 +0x272\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmt(0xc420678900, 0x2a76c40, 0xc420e59bf0, 0x0, 0x0, 0x0, 0x14dd98c1a27ac222, 0x1, 0xc420f69f80, 0xc42127c1f8, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1843 +0x62b\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmtInOpenTxn(0xc420678900, 0xc42127c000, 0x2a76c40, 0xc420e59bf0, 0x0, 0x0, 0x0, 0x14dd98c1a27ac222, 0x1, 0xc420f69f80, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1556 +0x6ce\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmtsInCurrentTxn(0xc420678900, 0xc42127c000, 0xc420c42400, 0x1, 0x1, 0x0, 0x1, 0x0, 0x2a76fc0, 0xc420da8300, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1127 +0x8c1\r\ngithub.com/cockroachdb/cockroach/pkg/sql.runTxnAttempt(0xc420678900, 0xc42127c000, 0xc420c42400, 0x1, 0x1, 0x0, 0x1, 0xc420330288, 0x1, 0x0, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1013 +0xc9\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execParsed.func1(0x7f5625754910, 0xc420c42480, 0xc420e5c160, 0xc420330288, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:821 +0x294\r\ngithub.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Exec(0xc420e5c160, 0x7f5625754910, 0xc420c42480, 0x101, 0xc420da4500, 0x201a852, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:722 +0xa8\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execParsed(0xc420678900, 0xc42127c000, 0xc420c42400, 0x1, 0x1, 0x0, 0x0, 0xc42001d900, 0x2)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:842 +0x47f\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execRequest(0xc420678900, 0xc42127c000, 0xc42127e000, 0x32, 0x0, 0x0, 0x32, 0x1000)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:727 +0x482\r\ngith" (spawn_id exp5) match glob pattern ":/# "? no

expect: does "\r\ngoroutine 618 [running]:\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.getStacks(0x200, 0x16753f99, 0x2f195e0, 0x10)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:824 +0xa7\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).outputLogEntry(0x2f1b2c0, 0xc400000004, 0x277a2d6, 0x16, 0x730, 0xc420e59e90, 0x2c)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:729 +0x576\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.addStructured(0x7f5625754910, 0xc420c42480, 0x7f5600000004, 0x2, 0x0, 0x0, 0xc42129a180, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/structured.go:161 +0x487\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.logDepth(0x7f5625754910, 0xc420c42480, 0x1, 0xc400000004, 0x0, 0x0, 0xc42129a180, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:52 +0x8c\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.Fatal(0x7f5625754910, 0xc420c42480, 0xc42129a180, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:155 +0x76\r\ngithub.com/cockroachdb/cockroach/pkg/sql/parser.glob..func142(0xc42127c248, 0xc4203304c0, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/parser/builtins.go:1840 +0x117\r\ngithub.com/cockroachdb/cockroach/pkg/sql/parser.(*FuncExpr).Eval(0xc420d20d80, 0xc42127c248, 0x1, 0xc4203304b0, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/parser/eval.go:2750 +0x232\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*renderNode).renderRow(0xc4212327e0, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/render.go:559 +0x77\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*renderNode).Next(0xc4212327e0, 0x7f5625754910, 0xc420c42480, 0xc42127c1f8, 0xc4212327e0, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/render.go:112 +0xc7\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execClassic(0xc420678900, 0xc42127c1f8, 0x2a75680, 0xc4212327e0, 0x2a7f040, 0xc420da8300, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1713 +0x272\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmt(0xc420678900, 0x2a76c40, 0xc420e59bf0, 0x0, 0x0, 0x0, 0x14dd98c1a27ac222, 0x1, 0xc420f69f80, 0xc42127c1f8, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1843 +0x62b\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmtInOpenTxn(0xc420678900, 0xc42127c000, 0x2a76c40, 0xc420e59bf0, 0x0, 0x0, 0x0, 0x14dd98c1a27ac222, 0x1, 0xc420f69f80, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1556 +0x6ce\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmtsInCurrentTxn(0xc420678900, 0xc42127c000, 0xc420c42400, 0x1, 0x1, 0x0, 0x1, 0x0, 0x2a76fc0, 0xc420da8300, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1127 +0x8c1\r\ngithub.com/cockroachdb/cockroach/pkg/sql.runTxnAttempt(0xc420678900, 0xc42127c000, 0xc420c42400, 0x1, 0x1, 0x0, 0x1, 0xc420330288, 0x1, 0x0, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1013 +0xc9\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execParsed.func1(0x7f5625754910, 0xc420c42480, 0xc420e5c160, 0xc420330288, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:821 +0x294\r\ngithub.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Exec(0xc420e5c160, 0x7f5625754910, 0xc420c42480, 0x101, 0xc420da4500, 0x201a852, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:722 +0xa8\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execParsed(0xc420678900, 0xc42127c000, 0xc420c42400, 0x1, 0x1, 0x0, 0x0, 0xc42001d900, 0x2)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:842 +0x47f\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execRequest(0xc420678900, 0xc42127c000, 0xc42127e000, 0x32, 0x0, 0x0, 0x32, 0x1000)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:727 +0x482\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).ExecuteStatements(0xc420678900, 0xc42127c000, 0xc42127e000, 0x32, 0x0, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:609 +0x129\r\ngithub.com/cockroachdb/cockroach/pkg/sql/pgwire.(*v3Conn).handleSimpleQuery(0xc420da8300, 0xc420da8328, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/v3.go:578 +0x148\r\ngithub.com/cockroachdb/cockroach/pkg/sql/pgwire.(*v3Conn).serve(0xc420da8300, 0x7f5625754650, 0xc420e59a10, 0xc420330110, 0x5400, 0xc420148168, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/v3.go:496 +0x8bb\r\ngithub.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).ServeConn(0xc420148000, 0x7f5625754910, 0xc420c42300, 0x2a7f220, 0xc420d5f7c0, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/server.go:421 +0x96c\r\ngithub.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func9.1(0x2a7f220, 0xc420d5f7c0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:718 +0x149\r\ngithub.com/cockroachdb/cockroach/pkg/util/netutil.(*Server).ServeWith.func1(0xc4206721b0, 0x7f5625754650, 0xc420544000, 0xc420482000, 0x2a7f220, 0xc420d5f7c0, 0xc420de2cf0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/netutil/net.go:140 +0xa9\r\ncreated by github.com/cockroachdb/cockroach/pkg/util/netutil.(*Server).ServeWith\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/netutil/net.go:142 +0x267\r\n:/# " (spawn_id exp5) match glob pattern ":/# "? yes
expect: set expect_out(0,string) ":/# "
expect: set expect_out(spawn_id) "exp5"
expect: set expect_out(buffer) "\r\ngoroutine 618 [running]:\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.getStacks(0x200, 0x16753f99, 0x2f195e0, 0x10)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:824 +0xa7\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).outputLogEntry(0x2f1b2c0, 0xc400000004, 0x277a2d6, 0x16, 0x730, 0xc420e59e90, 0x2c)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:729 +0x576\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.addStructured(0x7f5625754910, 0xc420c42480, 0x7f5600000004, 0x2, 0x0, 0x0, 0xc42129a180, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/structured.go:161 +0x487\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.logDepth(0x7f5625754910, 0xc420c42480, 0x1, 0xc400000004, 0x0, 0x0, 0xc42129a180, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:52 +0x8c\r\ngithub.com/cockroachdb/cockroach/pkg/util/log.Fatal(0x7f5625754910, 0xc420c42480, 0xc42129a180, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:155 +0x76\r\ngithub.com/cockroachdb/cockroach/pkg/sql/parser.glob..func142(0xc42127c248, 0xc4203304c0, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/parser/builtins.go:1840 +0x117\r\ngithub.com/cockroachdb/cockroach/pkg/sql/parser.(*FuncExpr).Eval(0xc420d20d80, 0xc42127c248, 0x1, 0xc4203304b0, 0x1, 0x1)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/parser/eval.go:2750 +0x232\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*renderNode).renderRow(0xc4212327e0, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/render.go:559 +0x77\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*renderNode).Next(0xc4212327e0, 0x7f5625754910, 0xc420c42480, 0xc42127c1f8, 0xc4212327e0, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/render.go:112 +0xc7\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execClassic(0xc420678900, 0xc42127c1f8, 0x2a75680, 0xc4212327e0, 0x2a7f040, 0xc420da8300, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1713 +0x272\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmt(0xc420678900, 0x2a76c40, 0xc420e59bf0, 0x0, 0x0, 0x0, 0x14dd98c1a27ac222, 0x1, 0xc420f69f80, 0xc42127c1f8, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1843 +0x62b\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmtInOpenTxn(0xc420678900, 0xc42127c000, 0x2a76c40, 0xc420e59bf0, 0x0, 0x0, 0x0, 0x14dd98c1a27ac222, 0x1, 0xc420f69f80, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1556 +0x6ce\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execStmtsInCurrentTxn(0xc420678900, 0xc42127c000, 0xc420c42400, 0x1, 0x1, 0x0, 0x1, 0x0, 0x2a76fc0, 0xc420da8300, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1127 +0x8c1\r\ngithub.com/cockroachdb/cockroach/pkg/sql.runTxnAttempt(0xc420678900, 0xc42127c000, 0xc420c42400, 0x1, 0x1, 0x0, 0x1, 0xc420330288, 0x1, 0x0, ...)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1013 +0xc9\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execParsed.func1(0x7f5625754910, 0xc420c42480, 0xc420e5c160, 0xc420330288, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:821 +0x294\r\ngithub.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Exec(0xc420e5c160, 0x7f5625754910, 0xc420c42480, 0x101, 0xc420da4500, 0x201a852, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:722 +0xa8\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execParsed(0xc420678900, 0xc42127c000, 0xc420c42400, 0x1, 0x1, 0x0, 0x0, 0xc42001d900, 0x2)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:842 +0x47f\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).execRequest(0xc420678900, 0xc42127c000, 0xc42127e000, 0x32, 0x0, 0x0, 0x32, 0x1000)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:727 +0x482\r\ngithub.com/cockroachdb/cockroach/pkg/sql.(*Executor).ExecuteStatements(0xc420678900, 0xc42127c000, 0xc42127e000, 0x32, 0x0, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:609 +0x129\r\ngithub.com/cockroachdb/cockroach/pkg/sql/pgwire.(*v3Conn).handleSimpleQuery(0xc420da8300, 0xc420da8328, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/v3.go:578 +0x148\r\ngithub.com/cockroachdb/cockroach/pkg/sql/pgwire.(*v3Conn).serve(0xc420da8300, 0x7f5625754650, 0xc420e59a10, 0xc420330110, 0x5400, 0xc420148168, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/v3.go:496 +0x8bb\r\ngithub.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).ServeConn(0xc420148000, 0x7f5625754910, 0xc420c42300, 0x2a7f220, 0xc420d5f7c0, 0x0, 0x0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/server.go:421 +0x96c\r\ngithub.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func9.1(0x2a7f220, 0xc420d5f7c0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:718 +0x149\r\ngithub.com/cockroachdb/cockroach/pkg/util/netutil.(*Server).ServeWith.func1(0xc4206721b0, 0x7f5625754650, 0xc420544000, 0xc420482000, 0x2a7f220, 0xc420d5f7c0, 0xc420de2cf0)\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/netutil/net.go:140 +0xa9\r\ncreated by github.com/cockroachdb/cockroach/pkg/util/netutil.(*Server).ServeWith\r\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/netutil/net.go:142 +0x267\r\n:/# "
github.com/cockroachdb/cockroach/pkg/sql.(*Executor).ExecuteStatements(0xc420678900, 0xc42127c000, 0xc42127e000, 0x32, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:609 +0x129
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*v3Conn).handleSimpleQuery(0xc420da8300, 0xc420da8328, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/v3.go:578 +0x148
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*v3Conn).serve(0xc420da8300, 0x7f5625754650, 0xc420e59a10, 0xc420330110, 0x5400, 0xc420148168, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/v3.go:496 +0x8bb
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).ServeConn(0xc420148000, 0x7f5625754910, 0xc420c42300, 0x2a7f220, 0xc420d5f7c0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/server.go:421 +0x96c
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func9.1(0x2a7f220, 0xc420d5f7c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:718 +0x149
github.com/cockroachdb/cockroach/pkg/util/netutil.(*Server).ServeWith.func1(0xc4206721b0, 0x7f5625754650, 0xc420544000, 0xc420482000, 0x2a7f220, 0xc420d5f7c0, 0xc420de2cf0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/netutil/net.go:140 +0xa9
created by github.com/cockroachdb/cockroach/pkg/util/netutil.(*Server).ServeWith
	/go/src/github.com/cockroachdb/cockroach/pkg/util/netutil/net.go:142 +0x267
send: sending "echo $?\r" to { exp5 }

expect: does "" (spawn_id exp5) match glob pattern "255"? no
:/# echo $?
255

expect: does "echo $?\r\n255\r\n:/# " (spawn_id exp5) match glob pattern "255"? yes
expect: set expect_out(0,string) "255"
expect: set expect_out(spawn_id) "exp5"
expect: set expect_out(buffer) "echo $?\r\n255"

expect: does "\r\n:/# " (spawn_id exp5) match glob pattern ":/# "? yes
expect: set expect_out(0,string) ":/# "
expect: set expect_out(spawn_id) "exp5"
expect: set expect_out(buffer) "\r\n:/# "
:/# 
.170823 22:03:28.406945966 EXPECT TEST: END TEST
system(echo; echo $(date '+.%y%m%d %H:%M:%S.%N') EXPECT TEST: 'END TEST' | tee -a logs/expect-cmd.log) = 2
system(find logs -exec chmod a+rw '{}' \;) = 2

.170823 22:03:28.436671769 EXPECT TEST: START TEST: Check that a server started with only in-memory stores and no --log-dir automatically logs to stderr.
system(echo; echo $(date '+.%y%m%d %H:%M:%S.%N') EXPECT TEST: 'START TEST: Check that a server started with only in-memory stores and no --log-dir automatically logs to stderr.' | tee -a logs/expect-cmd.log) = 2
system(find logs -exec chmod a+rw '{}' \;) = 2
send: sending "/cockroach/cockroach start --insecure --store=type=mem,size=1GiB\r" to { exp5 }

expect: does "" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --ins" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --insec" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --insecure" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --insecure -" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --insecure --s" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --insecure --sto" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --insecure --store" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --insecure --store=t" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --insecure --store=typ" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --insecure --store=type=" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --insecure --store=type=me" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --insecure --store=type=mem,s" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --insecure --store=type=mem,siz" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --insecure --store=type=mem,size" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --insecure --store=type=mem,size=1" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --insecure --store=type=mem,size=1G" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --insecure --store=type=mem,size=1Gi" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no
/cockroach/cockroach start --insecure --store=type=mem,size=1GiB

expect: does "/cockroach/cockroach start --insecure --store=type=mem,size=1GiB\r\n" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no
W170823 22:03:28.505932 1 cli/start.go:601  RUNNING IN INSECURE MODE!

- Your cluster is open for any client that can access <all your IP addresses>.
- Any user, even root, can log in without providing a password.
- Any user, connecting as root, can read or write any data in your cluster.
- There is no network encryption nor authentication, and thus no confidentiality.

Check out how to secure your cluster: https://www.cockroachlabs.com/docs/stable/secure-a-cluster.html

expect: does "/cockroach/cockroach start --insecure --store=type=mem,size=1GiB\r\nW170823 22:03:28.505932 1 cli/start.go:601  RUNNING IN INSECURE MODE!\r\n\r\n- Your cluster is open for any client that can access <all your IP addresses>.\r\n- Any user, even root, can log in without providing a password.\r\n- Any user, connecting as root, can read or write any data in your cluster.\r\n- There is no network encryption nor authentication, and thus no confidentiality.\r\n\r\nCheck out how to secure your cluster: https://www.cockroachlabs.com/docs/stable/secure-a-cluster.html\r\n" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --insecure --store=type=mem,size=1GiB\r\nW170823 22:03:28.505932 1 cli/start.go:601  RUNNING IN INSECURE MODE!\r\n\r\n- Your cluster is open for any client that can access <all your IP addresses>.\r\n- Any user, even root, can log in without providing a password.\r\n- Any user, connecting as root, can read or write any data in your cluster.\r\n- There is no network encryption nor authentication, and thus no confidentiality.\r\n\r\nCheck out how to secure your cluster: https://www.cockroachlabs.com/docs/stable/secure-a-cluster.html\r\nI170823 22:03:28.506016 1 cli/start.go:607  CockroachDB CCL 42e183b (linux amd64, built 2017/08/23 22:00:38, go1.8.3)\r\n" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no
I170823 22:03:28.506016 1 cli/start.go:607  CockroachDB CCL 42e183b (linux amd64, built 2017/08/23 22:00:38, go1.8.3)
I170823 22:03:28.607273 1 server/config.go:284  can't read available memory from cgroups (open /sys/fs/cgroup/memory/memory.limit_in_bytes: no such file or directory), using system memory 28 GiB instead
I170823 22:03:28.607329 1 server/config.go:419  system total memory: 28 GiB

expect: does "/cockroach/cockroach start --insecure --store=type=mem,size=1GiB\r\nW170823 22:03:28.505932 1 cli/start.go:601  RUNNING IN INSECURE MODE!\r\n\r\n- Your cluster is open for any client that can access <all your IP addresses>.\r\n- Any user, even root, can log in without providing a password.\r\n- Any user, connecting as root, can read or write any data in your cluster.\r\n- There is no network encryption nor authentication, and thus no confidentiality.\r\n\r\nCheck out how to secure your cluster: https://www.cockroachlabs.com/docs/stable/secure-a-cluster.html\r\nI170823 22:03:28.506016 1 cli/start.go:607  CockroachDB CCL 42e183b (linux amd64, built 2017/08/23 22:00:38, go1.8.3)\r\nI170823 22:03:28.607273 1 server/config.go:284  can't read available memory from cgroups (open /sys/fs/cgroup/memory/memory.limit_in_bytes: no such file or directory), using system memory 28 GiB instead\r\nI170823 22:03:28.607329 1 server/config.go:419  system total memory: 28 GiB\r\n" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no
I170823 22:03:28.607378 1 server/config.go:421  server configuration:
max offset                500000000
cache size                6.9 GiB
SQL memory pool size      6.9 GiB
scan interval             10m0s
scan max idle time        200ms
metrics sample interval   10s
time until store dead     5m0s
event log enabled         true
linearizable              false
I170823 22:03:28.607477 36 cli/start.go:344  starting cockroach node

expect: does "/cockroach/cockroach start --insecure --store=type=mem,size=1GiB\r\nW170823 22:03:28.505932 1 cli/start.go:601  RUNNING IN INSECURE MODE!\r\n\r\n- Your cluster is open for any client that can access <all your IP addresses>.\r\n- Any user, even root, can log in without providing a password.\r\n- Any user, connecting as root, can read or write any data in your cluster.\r\n- There is no network encryption nor authentication, and thus no confidentiality.\r\n\r\nCheck out how to secure your cluster: https://www.cockroachlabs.com/docs/stable/secure-a-cluster.html\r\nI170823 22:03:28.506016 1 cli/start.go:607  CockroachDB CCL 42e183b (linux amd64, built 2017/08/23 22:00:38, go1.8.3)\r\nI170823 22:03:28.607273 1 server/config.go:284  can't read available memory from cgroups (open /sys/fs/cgroup/memory/memory.limit_in_bytes: no such file or directory), using system memory 28 GiB instead\r\nI170823 22:03:28.607329 1 server/config.go:419  system total memory: 28 GiB\r\nI170823 22:03:28.607378 1 server/config.go:421  server configuration:\r\nmax offset                500000000\r\ncache size                6.9 GiB\r\nSQL memory pool size      6.9 GiB\r\nscan interval             10m0s" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no
I170823 22:03:28.607511 36 cli/start.go:346  using local environment variables: COCKROACH_SQL_CLI_HISTORY=cockroach_sql_history, COCKROACH_INSECURE=true

expect: does "/cockroach/cockroach start --insecure --store=type=mem,size=1GiB\r\nW170823 22:03:28.505932 1 cli/start.go:601  RUNNING IN INSECURE MODE!\r\n\r\n- Your cluster is open for any client that can access <all your IP addresses>.\r\n- Any user, even root, can log in without providing a password.\r\n- Any user, connecting as root, can read or write any data in your cluster.\r\n- There is no network encryption nor authentication, and thus no confidentiality.\r\n\r\nCheck out how to secure your cluster: https://www.cockroachlabs.com/docs/stable/secure-a-cluster.html\r\nI170823 22:03:28.506016 1 cli/start.go:607  CockroachDB CCL 42e183b (linux amd64, built 2017/08/23 22:00:38, go1.8.3)\r\nI170823 22:03:28.607273 1 server/config.go:284  can't read available memory from cgroups (open /sys/fs/cgroup/memory/memory.limit_in_bytes: no such file or directory), using system memory 28 GiB instead\r\nI170823 22:03:28.607329 1 server/config.go:419  system total memory: 28 GiB\r\nI170823 22:03:28.607378 1 server/config.go:421  server configuration:\r\nmax offset                500000000\r\ncache size                6.9 GiB\r\nSQL memory pool size      6.9 GiB\r\nscan interval             10m0s\r\nscan max idle time        200ms\r\nmetrics sample interval   10s\r\ntime until store dead     5m0s\r\nevent log enabled         true\r\nlinearizable              false\r\n" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --insecure --store=type=mem,size=1GiB\r\nW170823 22:03:28.505932 1 cli/start.go:601  RUNNING IN INSECURE MODE!\r\n\r\n- Your cluster is open for any client that can access <all your IP addresses>.\r\n- Any user, even root, can log in without providing a password.\r\n- Any user, connecting as root, can read or write any data in your cluster.\r\n- There is no network encryption nor authentication, and thus no confidentiality.\r\n\r\nCheck out how to secure your cluster: https://www.cockroachlabs.com/docs/stable/secure-a-cluster.html\r\nI170823 22:03:28.506016 1 cli/start.go:607  CockroachDB CCL 42e183b (linux amd64, built 2017/08/23 22:00:38, go1.8.3)\r\nI170823 22:03:28.607273 1 server/config.go:284  can't read available memory from cgroups (open /sys/fs/cgroup/memory/memory.limit_in_bytes: no such file or directory), using system memory 28 GiB instead\r\nI170823 22:03:28.607329 1 server/config.go:419  system total memory: 28 GiB\r\nI170823 22:03:28.607378 1 server/config.go:421  server configuration:\r\nmax offset                500000000\r\ncache size                6.9 GiB\r\nSQL memory pool size      6.9 GiB\r\nscan interval             10m0s\r\nscan max idle time        200ms\r\nmetrics sample interval   10s\r\ntime until store dead     5m0s\r\nevent log enabled         true\r\nlinearizable              false\r\nI170823 22:03:28.607477 36 cli/start.go:344  starting cockroach node\r\n" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --insecure --store=type=mem,size=1GiB\r\nW170823 22:03:28.505932 1 cli/start.go:601  RUNNING IN INSECURE MODE!\r\n\r\n- Your cluster is open for any client that can access <all your IP addresses>.\r\n- Any user, even root, can log in without providing a password.\r\n- Any user, connecting as root, can read or write any data in your cluster.\r\n- There is no network encryption nor authentication, and thus no confidentiality.\r\n\r\nCheck out how to secure your cluster: https://www.cockroachlabs.com/docs/stable/secure-a-cluster.html\r\nI170823 22:03:28.506016 1 cli/start.go:607  CockroachDB CCL 42e183b (linux amd64, built 2017/08/23 22:00:38, go1.8.3)\r\nI170823 22:03:28.607273 1 server/config.go:284  can't read available memory from cgroups (open /sys/fs/cgroup/memory/memory.limit_in_bytes: no such file or directory), using system memory 28 GiB instead\r\nI170823 22:03:28.607329 1 server/config.go:419  system total memory: 28 GiB\r\nI170823 22:03:28.607378 1 server/config.go:421  server configuration:\r\nmax offset                500000000\r\ncache size                6.9 GiB\r\nSQL memory pool size      6.9 GiB\r\nscan interval             10m0s\r\nscan max idle time        200ms\r\nmetrics sample interval   10s\r\ntime until store dead     5m0s\r\nevent log enabled         true\r\nlinearizable              false\r\nI170823 22:03:28.607477 36 cli/start.go:344  starting cockroach node\r\nI170823 22:03:28.607511 36 cli/start.go:346  using local environment variables: COCKROACH_SQL_CLI_HISTORY=cockroach_sql_history, COCKROACH_INSECURE=true\r\n" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no
I170823 22:03:28.615597 36 server/config.go:522  [n?] 1 storage engine initialized

expect: does "/cockroach/cockroach start --insecure --store=type=mem,size=1GiB\r\nW170823 22:03:28.505932 1 cli/start.go:601  RUNNING IN INSECURE MODE!\r\n\r\n- Your cluster is open for any client that can access <all your IP addresses>.\r\n- Any user, even root, can log in without providing a password.\r\n- Any user, connecting as root, can read or write any data in your cluster.\r\n- There is no network encryption nor authentication, and thus no confidentiality.\r\n\r\nCheck out how to secure your cluster: https://www.cockroachlabs.com/docs/stable/secure-a-cluster.html\r\nI170823 22:03:28.506016 1 cli/start.go:607  CockroachDB CCL 42e183b (linux amd64, built 2017/08/23 22:00:38, go1.8.3)\r\nI170823 22:03:28.607273 1 server/config.go:284  can't read available memory from cgroups (open /sys/fs/cgroup/memory/memory.limit_in_bytes: no such file or directory), using system memory 28 GiB instead\r\nI170823 22:03:28.607329 1 server/config.go:419  system total memory: 28 GiB\r\nI170823 22:03:28.607378 1 server/config.go:421  server configuration:\r\nmax offset                500000000\r\ncache size                6.9 GiB\r\nSQL memory pool size      6.9 GiB\r\nscan interval             10m0s\r\nscan max idle time        200ms\r\nmetrics sample interval   10s\r\ntime until store dead     5m0s\r\nevent log enabled         true\r\nlinearizable              false\r\nI170823 22:03:28.607477 36 cli/start.go:344  starting cockroach node\r\nI170823 22:03:28.607511 36 cli/start.go:346  using local environment variables: COCKROACH_SQL_CLI_HISTORY=cockroach_sql_history, COCKROACH_INSECURE=true\r\nI170823 22:03:28.615597 36 server/config.go:522  [n?] 1 storage engine initialized\r\nI170823 22:03:28.615636 36 server/config.go:524  [n?] RocksDB cache size: 6.9 GiB" (spawn_id exp5) match glob pattern "CockroachDB node starting"? no

expect: does "/cockroach/cockroach start --insecure --store=type=mem,size=1GiB\r\nW170823 22:03:28.505932 1 cli/start.go:601  RUNNING IN INSECURE MODE!\r\n\r\n- Your cluster is open for any client that can access <all your IP addresses>.\r\n- Any user, even root, can log in without providing a password.\r\n- Any user, connecting as root, can read or write any data in your cluster.\r\n- There is no network encryption nor authentication, and thus no confidentiality.\r\n\r\nCheck out how to secure your cluster: https://www.cockroachlabs.com/docs/stable/secure-a-cluster.html\r\nI170823 22:03:28.506016 1 cli/start.go:607  CockroachDB CCL 42e183b (linux amd64, built 2017/08/23 22:00:38, go1.8.3)\r\nI170823 22:03:28.607273 1 server/config.go:284  can't read available memory from cgroups (open /sys/fs/cgroup/memory/memory.limit_in_bytes: no such file or directory), using system memory 28 GiB instead\r\nI170823 22:03:28.607329 1 server/config.go:419  system total memory: 28 GiB\r\nI170823 22:03:28.607378 1 server/config.go:421  server configuration:\r\nmax offset                500000000\r\ncache size                6.9 GiB\r\nSQL memory pool size      6.9 GiB\r\nscan interval             10m0s\r\nscan max idle time        200ms\r\nmetrics sample interval   10s\r\ntime until store dead     5
```
Please assign, take a look and update the issue accordingly.

--- a/pkg/cli/interactive_tests/common.tcl
+++ b/pkg/cli/interactive_tests/common.tcl
@@ -35,7 +35,12 @@ proc report {text} {
     # Docker is obnoxious in that it doesn't support setting `umask`.
     # Also CockroachDB doesn't honor umask anyway.
     # So we simply come after the fact and adjust the permissions.
-    system "find logs -exec chmod a+rw '{}' \\;"
+    #
+    # The find may race with a cockroach process shutting down in the
+    # background; cockroach might be deleting files as they are being
+    # found, causing chmod to not find its target file. We ignore
+    # these errors.
+    system "find logs -exec chmod a+rw '{}' \\; || true"
 }
 
 # Catch signals

--- a/pkg/cli/interactive_tests/test_log_flags.tcl
+++ b/pkg/cli/interactive_tests/test_log_flags.tcl
@@ -12,7 +12,7 @@ eexpect ":/# "
 # errors when parsing flags in that context cause the (test) process
 # to exit entirely (it has errorHandling set to ExitOnError).
 
-# Check that log files are created by default in the store directory.
+start_test "Check that log files are created by default in the store directory."
 send "$argv start --insecure --store=path=logs/mystore\r"
 eexpect "node starting"
 interrupt
@@ -20,8 +20,9 @@ eexpect ":/# "
 send "ls logs/mystore/logs\r"
 eexpect "cockroach.log"
 eexpect ":/# "
+end_test
 
-# Check that an empty `-log-dir` disables file logging.
+start_test "Check that an empty -log-dir disables file logging."
 send "$argv start --insecure --store=path=logs/mystore2 --log-dir=\r"
 eexpect "node starting"
 interrupt
@@ -29,19 +30,22 @@ eexpect ":/# "
 send "ls logs/mystore2/logs 2>/dev/null | wc -l\r"
 eexpect "0"
 eexpect ":/# "
+end_test
 
-# Check that leading tildes are properly rejected.
+start_test "Check that leading tildes are properly rejected."
 send "$argv start --insecure -s=path=logs/db --log-dir=\~/blah\r"
 eexpect "log directory cannot start with '~'"
 eexpect ":/# "
+end_test
 
-# Check that the user can override.
+start_test "Check that the user can override."
 send "$argv start --insecure -s=path=logs/db --log-dir=logs/blah/\~/blah\r"
 eexpect "logs: *blah/~/blah"
 interrupt
 eexpect ":/# "
+end_test
 
-# Check that TRUE and FALSE are valid values for the severity flags.
+start_test "Check that TRUE and FALSE are valid values for the severity flags."
 send "$argv start --insecure -s=path=logs/db --logtostderr=false\r"
 eexpect "node starting"
 interrupt
@@ -57,6 +61,7 @@ eexpect ":/# "
 send "$argv start --insecure -s=path=logs/db --logtostderr=cantparse\r"
 eexpect "parsing \"cantparse\": invalid syntax"
 eexpect ":/# "
+end_test
 
 send "exit 0\r"
 eexpect eof

--- a/pkg/cli/interactive_tests/test_sql_mem_monitor.tcl
+++ b/pkg/cli/interactive_tests/test_sql_mem_monitor.tcl
@@ -58,12 +58,16 @@ send "select 1;\r"
 eexpect "1 row"
 eexpect root@
 
+start_test "Ensure that memory over-allocation without monitoring crashes the server"
 # Now try to run a large-ish query on the client.
 # The query is a 4-way cross-join on information_schema.columns,
 # resulting in ~8 million rows loaded into memory when run on an
 # empty database.
 send "set database=information_schema;\r"
 eexpect root@
+# Disable query distribution to force in-memory computation.
+send "set distsql=off;\r"
+eexpect SET
 send "select * from columns as a, columns as b, columns as c, columns as d limit 10;\r"
 
 # Check that the query crashed the server
@@ -81,7 +85,9 @@ eexpect ":/# "
 set spawn_id $client_spawn_id
 eexpect "bad connection"
 eexpect root@
+end_test
 
+start_test "Ensure that memory monitoring prevents crashes"
 # Re-launch a server with relatively lower limit for SQL memory
 set spawn_id $shell_spawn_id
 send "$argv start --insecure --max-sql-memory=150K --no-redirect-stderr -s=path=logs/db \r"
@@ -102,5 +108,15 @@ eexpect root@
 send "select 1;\r"
 eexpect "1 row"
 eexpect root@
+end_test
 
-# We just terminate, this will kill both server and client.
+interrupt
+eexpect eof
+
+set spawn_id $shell_spawn_id
+interrupt
+interrupt
+eexpect ":/# "
+send "exit\r"
+eexpect eof
+

